src/spox/_fields.py

```diff
@@ -59,13 +59,15 @@ def __post_init__(self):
                     )
 
     @classmethod
-    def _get_field_type(cls, field) -> VarFieldKind:
+    def _get_field_type(cls, field: Field) -> VarFieldKind:
         """Access the kind of the field (single, optional, variadic) based on its type annotation."""
-        if field.type == Var:
+        # The field may be unannotated as per
+        # from __future__ import annotations
+        if field.type in [Var, "Var"]:
             return VarFieldKind.SINGLE
-        elif field.type == Optional[Var]:
+        elif field.type in [Optional[Var], "Optional[Var]"]:
             return VarFieldKind.OPTIONAL
-        elif field.type == Sequence[Var]:
+        elif field.type in [Sequence[Var], "Sequence[Var]"]:
```

Comment:
I'm afraid this is not the ideal fix. We should instead use `typing.get_type_hints` to evaluate the hints properly. E.g.:

```python
typing.get_type_hints(cls)[field.name] == Optional[Var]
```

Note that this works:
```python
assert None | Var == Optional[Var]
```